flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -132,7 +143,61 @@ private static JobExceptionsInfo createJobExceptionsInfo(
             }
         }
 
+        final ErrorInfo rootCause = executionGraph.getFailureInfo();
+        return new JobExceptionsInfoWithHistory(
+                rootCause.getException().getOriginalErrorClassName(),
+                rootCause.getExceptionAsString(),
+                rootCause.getTimestamp(),
+                taskExceptionList,
+                truncated,
+                createJobExceptionHistory(
+                        executionGraphInfo.getExceptionHistory(), exceptionToReportMaxSize));
+    }
+
+    static JobExceptionsInfoWithHistory.JobExceptionHistory createJobExceptionHistory(
+            BoundedFIFOQueue<ExceptionHistoryEntry> historyEntries, int limit) {
+        // we need to reverse the history to have a stable result when doing paging on it
+        final List<ExceptionHistoryEntry> reversedHistoryEntries = new ArrayList<>();
+        Iterables.addAll(reversedHistoryEntries, historyEntries);
+        Collections.reverse(reversedHistoryEntries);
+
+        List<JobExceptionsInfo> exceptionHistoryEntries =
+                reversedHistoryEntries.stream()
+                        .limit(limit)
+                        .map(JobExceptionsHandler::createJobExceptionInfo)
+                        .collect(Collectors.toList());
+
+        return new JobExceptionsInfoWithHistory.JobExceptionHistory(
+                exceptionHistoryEntries,
+                historyEntries.getElementCount(),
```

Comment:
The exception count gives an understanding on how many exceptions where caught in total since we're limiting the history through a parameter. Thinking about it once more: It might not be useful for long-running jobs, does it?

Reply:
My initial motivation was to make it more transparent that the exceptions listed are just the most recent ones. Removing the `total` count from the response object might make the user assume that these are the only exceptions that were caught.

Reply:
What about renaming `exception-history` into `most-recent-exceptions`, instead?

Reply:
My immediate assumption when hearing "recent" is that it depends on time in some form, which doesn't apply here.
I think "exception-history" is fine.

Reply:
I had a chat with @NicoK about it. He agrees that (on the UI side) it's not necessarily useful to have the over all count of exceptions presented as it might grow for long-running jobs. Instead, we could introduce a `max-size-reached` flag to indicate that exception history's maximum size was reached. The UI would then just print `...` and have a refer to the corresponding configuration option and the logs for further analysis.

Reply:
The truncation information is not exposed anymore to avoid confusion. A static info text was added that pops up as soon as the history has some entries.